src/API.jl

```diff
@@ -972,11 +975,13 @@ function precompile(ctx::Context; internal_call::Bool=false, io::IO=stderr)
     failed_deps = Base.PkgId[]
     skipped_deps = Base.PkgId[]
 
-    print_lock = stdout isa Base.LibuvStream ? stdout.lock : ReentrantLock()
+    print_lock= (stdout isa Base.LibuvStream ? stdout.lock : ReentrantLock())::ReentrantLock
     first_started = Base.Event()
-    finished = false
-    should_exit = !fancy_print # exit print loop immediately if not fancy printing
-    interrupted = false
+    # We make these Refs so they do not have to be boxed
```

Comment:
Probably prettier if you change it to
```julia
finished::Bool = false
should_exit::Bool = !fancy_print
interrupted::Bool = false
show_report::Bool = true
```

Reply:
Yes, would be prettier but doesn't remove the boxes from what I can see. I've found type asserts to be quite ineffective when it comes to boxes.

Reply:
It may not remove the boxes but with this change they don't matter. But either way is fine.

Reply:
Alright, I changed it to the type asserts.